oteapi_dlite/strategies/mapping.py

```diff
@@ -326,26 +338,46 @@ def fetch_and_populate_graph(
         PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
         PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
         PREFIX fno: <https://w3id.org/function/ontology#>
+        PREFIX emmo: <http://emmo.info/domain-mappings#>
+        PREFIX oteio: <http://emmo.info/oteio#>
 
-        SELECT ?subject ?predicate ?object
+        SELECT DISTINCT ?subject ?predicate ?object
         WHERE {{
-           GRAPH <{graph_uri}> {{
+        GRAPH <{graph_uri}> {{
+
+            # Match all subclasses of the parent class
+            ?subclass rdfs:subClassOf* <{parent_node}> .
+
+            # Retrieve all relevant triples for these subclasses and their individuals
+            {{
+            # Subclasses themselves and their relationships
+            ?subclass ?predicate ?object .
+            BIND(?subclass AS ?subject)
+            }} UNION {{
+            # Individuals of these subclasses and their properties
+            ?subject rdf:type ?subclass .
             ?subject ?predicate ?object .
-            ?subject rdfs:subClassOf* <{parent_node}> .
+            }}
+
+            # Ensure subject, predicate, and object are not empty
+            FILTER(BOUND(?subject) && BOUND(?predicate) && BOUND(?object))
+
+            # Filter by the specific predicates
             FILTER (?predicate IN (
-                rdfs:subClassOf,
```

Comment:
I think we need the `subClassOf` and perhaps even add `hasSubClass`. Sometime we will need to traverse the graph up and down the inheritance levels.

Reply:
ok will add it back